src/jarabe/journal/journalactivity.py

```diff
@@ -257,20 +304,91 @@ def _setup_main_view(self):
+                                self.__title_edit_finished_cb)
+        self._list_view_project.connect('selection-changed',
+                                self.__selection_changed_cb)
+        self._list_view_project.connect('project-view-activate',
+                                self.project_view_activated_cb)
+        return self._list_view_project
+
+    def _create_volumes_toolbar(self):
         self._volumes_toolbar = VolumesToolbar()
         self._volumes_toolbar.connect('volume-changed',
                                       self.__volume_changed_cb)
         self._volumes_toolbar.connect('volume-error', self.volume_error_cb)
-        self._main_view.pack_start(self._volumes_toolbar, False, True, 0)
+        return self._volumes_toolbar
+
+    def _setup_main_view(self):
+        self._main_toolbox = MainToolbox()
+        self._edit_toolbox = EditToolbox(self)
+        self._main_view = Gtk.VBox()
+
+        add_new_box = self._create_add_new_entry()
+        self._main_view.pack_start(add_new_box, False, True, style.DEFAULT_SPACING)
+        self._main_view.set_can_focus(True)
+
+        list_view = self._create_list_view()
+        self._main_view.pack_start(list_view, True, True, 0)
+        list_view.show_all()
+
+        volumes_toolbar = self._create_volumes_toolbar()
+        self._main_view.pack_start(volumes_toolbar, False, True, 0)
 
         self._main_toolbox.connect('query-changed', self._query_changed_cb)
+
         self._main_toolbox.search_entry.connect('icon-press',
                                                 self.__search_icon_pressed_cb)
         self._main_toolbox.set_mount_point(self._mount_point)
 
+    def _setup_project_view(self):
+        self._project_view = ProjectView()
+        add_new_box = self._create_add_new_entry_project()
+        add_new_box.show_all()
+        self._project_view.pack_start(add_new_box, False, True, style.DEFAULT_SPACING/3)
+        list_view = self._create_list_view_project()
+        self._project_view.pack_start(list_view, True, True, 0)
+        list_view.show()
+        
+    def get_list_view(self):
+        return self._list_view
+
+    def get_entry(self):
+        return self._entry
+
+    def project_view_activated_cb(self, list_view, metadata):
+        self.project_metadata = metadata
+        self._project_view.set_project_metadata(self.project_metadata)
+
+        self._project_view.connect('go-back-clicked', self.__go_back_clicked_cb)
+        self._main_view_active = False
+        self.get_list_view().set_projects_view_active(False)
+        logging.debug('project_view_activate signal handler')
+        self.set_canvas(self._project_view)
+        self._toolbox = self._main_toolbox
+        self.set_toolbar_box(self._toolbox)
+        self._toolbox.show()
+
+        query = {}
+        query['project_id'] = self.project_metadata['uid']
+        #query['mountpoints'] = ['/']
+        self._list_view_project.update_with_query(query)
```

Comment:
Can we make the _list_view_project a responsability of the ProjectView?  So we give it to the ProjectView constructor, and the ProjectView deals with adding it to it's box, updating the query, etc.